### PR TITLE
Fixes #29716 - fix  deprecation warning

### PR DESCRIPTION
**Pull request by shiramax**
fix `Module#parent` deprecation warning

### CODE REVIEW

File: lib/fog/ovirt/compute/v4.rb [hunk 1]
```diff
@@ -104,7 +104,7 @@ def get_attr_value(value, opts)
             when OvirtSDK4::Console
               { :enabled => value.enabled }
             else
-              if value.class.respond_to?(:parent) && value.class.parent == OvirtSDK4
+              if value.class.respond_to?(:module_parent) && value.class.module_parent == OvirtSDK4
```
Comment (ShimShtein) [inline lib/fog/ovirt/compute/v4.rb:107]: Not the intention of this PR, but you can use `try`
```suggestion
              value.class.try(:module_parent) == OvirtSDK4
```

**Review by ShimShtein — CHANGES_REQUESTED** [review]

**Review by ShimShtein — APPROVED** [review]

File: lib/fog/ovirt/compute/v4.rb [hunk 2]
```diff
@@ -104,7 +104,7 @@ def get_attr_value(value, opts)
             when OvirtSDK4::Console
               { :enabled => value.enabled }
             else
-              if value.class.respond_to?(:parent) && value.class.parent == OvirtSDK4
+              value.class.try(:module_parent) == OvirtSDK4
```
Comment (ShimShtein) [inline lib/fog/ovirt/compute/v4.rb]:
```suggestion
              if value.class.try(:module_parent) == OvirtSDK4
```

**Review by ShimShtein — CHANGES_REQUESTED** [review]
Sorry, you still need the if statement, my bad

**Comment by ShimShtein** [comment]
Apparently `try` is rails thing. No support for that in pure ruby. Sorry for misleading.

**Review by ShimShtein — APPROVED** [review]
ACK, @orrabin, I see the failures are related to Rubocop in ruby 2.4, and not entirely related to this PR. Do we want to merge this PR regardless rubocop failures?

**Comment by orrabin** [comment]
I would go with either a PR to fix the rubocop errors first or disabling them if we don't want these cops.

**Comment by ShimShtein** [comment]
ACK. This makes sense

**Comment by ShimShtein** [comment]
Merged, thanks @shiramax!

**Pull request merged**